Docs: add architectural documentation (#46)
SanteonNL/orca: README.md

3. At this point, the Care Plan Service notifies the Task filler that a new Task is available.
4. The Task filler and placer now negotiate the Task details:
   1. Filler: if the filler needs more information, it adds a Questionnaire to the `Task.input`.
   2. Placer: responds by filling the Questionnare, adding a QuestionnaireResponse to the Task.output.
   3. Filler: verifies the QuestionnaireResponse either:
      - Accept the Task if it is able and willing to perform it.
      - Add another Questionnaire to the `Task.input` if it needs more information.
      - Reject the Task if it cannot/won't perform it.

Whenever a Task is created/updated at the Care Plan Service, it notifies the other Task participant.
The notification is then handled to perform the Task negotiation:
- Filler: notification is received by ORCA's *Orchestrator* and forwarded to the *Task Engine* (TODO).
  The *Task Engine* then decides whether to accept the Task, and if not, what additional information is required.
- Placer: notification is received by ORCA's *Orchestrator* and forwarded to the *Task Engine* (TODO).

>   The *Frontend* has a websocket connection to the *Task Engine* to receive updates on the Task status.

Reviewer: I think it's probably easier to use something like an `EventSource`  to stream data from the server to the client, we only need to receive data, and it's much easier to implement. 

```suggestion
  The *Frontend* is notified by the *Task Engine* when the Task status changes.
```
